src/trigger.c

```diff
@@ -0,0 +1,292 @@
+#include "common.h"
```

Comment:
Good idea renaming this file 👍 
Since git diff thinks this whole file is new I'm gonna review it as if you wrote everything, apologies if I ask you to change stuff you didn't write but they're worth doing now IMO